tiled/client/sync.py

```diff
@@ -0,0 +1,134 @@
+
+
+def _copy_array(source, dest):
+    num_blocks = (range(len(n)) for n in source.chunks)
+    # Loop over each block index --- e.g. (0, 0), (0, 1), (0, 2) ....
+    for block in itertools.product(*num_blocks):
+        array = source.read_block(block)
+        dest.write_block(array, block)
+
+
+def _copy_awkward(source, dest):
+    import awkward
+
+    array = source.read()
+    _form, _length, container = awkward.to_buffers(array)
+    dest.write(container)
+
+
+def _copy_sparse(source, dest):
+    num_blocks = (range(len(n)) for n in source.chunks)
+    # Loop over each block index --- e.g. (0, 0), (0, 1), (0, 2) ....
+    for block in itertools.product(*num_blocks):
+        array = source.read_block(block)
+        dest.write_block(array.coords, array.data, block)
+
+
+def _copy_table(source, dest):
+    for partition in range(source.structure().npartitions):
+        df = source.read_partition(partition)
+        dest.write_partition(df, partition)
+
+
+def _copy_container(source, dest):
+    for key, child_node in source.items():
+        original_data_sources = child_node.include_data_sources().data_sources()
+        num_data_sources = len(original_data_sources)
+        if num_data_sources == 0:
+            # A container with no data sources is just an organizational
+            # entity in the database.
+            if child_node.structure_family == StructureFamily.container:
+                data_sources = []
+            else:
+                raise ValueError(
+                    f"Unable to copy {child_node} which is a "
+                    f"{child_node.structure_family} but has no data sources."
+                )
+        elif num_data_sources == 1:
+            (original_data_source,) = original_data_sources
+            if original_data_source.management == Management.external:
+                data_sources = [original_data_source]
+            else:
+                if child_node.structure_family == StructureFamily.container:
+                    data_sources = []
+                else:
+                    data_sources = [
+                        DataSource(
+                            management=original_data_source.management,
+                            mimetype=original_data_source.mimetype,
+                            structure_family=original_data_source.structure_family,
+                            structure=original_data_source.structure,
+                        )
+                    ]
+        else:
+            # As of this writing this is impossible, but we anticipate that
+            # it may be added someday.
+            raise NotImplementedError(
+                "Multiple Data Sources in one Node is not supported."
+            )
+        node = dest.new(
+            key=key,
+            structure_family=child_node.structure_family,
+            data_sources=data_sources,
+            metadata=dict(child_node.metadata),
+            specs=child_node.specs,
+        )
```

Comment:
```suggestion
        try:
            node = dest.new(
                key=key,
                structure_family=child_node.structure_family,
                data_sources=data_sources,
                metadata=dict(child_node.metadata),
                specs=child_node.specs,
            )
        except tiled.client.util.ClientError as err:
            if err.response.status_code == httpx.codes.CONFLICT:
                # print('Skipped existing entry (or UUID conflict)')
                continue
            else:
                raise e
        
```
this is a crude and probably relatively non-performant, but functional way to deal with a user request to copy already-copied items.  also needs two imports:
```
from .utils import ClientError
import httpx 
```

Reply:
I think this comment just missed @genematx’s merge commit.

I can see the argument for a simple "skip conflicts" option. Let's do a little exploration of what options we may want in the future and then start with something simple.

Reply:
Would you make a separate follow-up PR out of this commit, @pbeaucage ?

Reply:
No worries!  See #737